mobile/src/main/java/org/openhab/habdroid/core/OpenHABVoiceService.java

```diff
@@ -34,68 +31,35 @@
  * This service handles voice commands and sends them to OpenHAB.
  * It will use the openHAB base URL if passed in the intent's extra.
  */
-public class OpenHABVoiceService extends Service implements ConnectionFactory.UpdateListener {
+public class OpenHABVoiceService extends IntentService {
     private static final String TAG = OpenHABVoiceService.class.getSimpleName();
 
-    private final List<Pair<String, Integer>> mPendingCommands = new ArrayList<>();
-
     public OpenHABVoiceService() {
+        super("OpenHABVoiceService");
     }
 
     @Override
-    public int onStartCommand(Intent intent, int flags, final int startId) {
+    protected void onHandleIntent(@Nullable Intent intent) {
         final String voiceCommand = extractVoiceCommand(intent);
-        boolean hasSentCommand = false;
-        if (!voiceCommand.isEmpty()) {
-            try {
-                Connection conn = ConnectionFactory.getUsableConnection();
-                if (conn != null) {
-                    sendItemCommand("VoiceCommand", voiceCommand, conn, startId);
-                } else {
-                    mPendingCommands.add(Pair.create(voiceCommand, startId));
-                    ConnectionFactory.addListener(this);
-                }
-                hasSentCommand = true;
-            } catch (ConnectionException e) {
-                Log.w(TAG, "Couldn't determine OpenHAB URL", e);
-                Toast.makeText(this,
-                        R.string.error_couldnt_determine_openhab_url, Toast.LENGTH_SHORT)
-                        .show();
-            }
-        }
-        if (!hasSentCommand) {
-            stopSelf(startId);
+        if (voiceCommand.isEmpty()) {
+            return;
         }
-        return START_NOT_STICKY;
-    }
-
-    @Override
-    public void onDestroy() {
-        super.onDestroy();
-        ConnectionFactory.removeListener(this);
-    }
 
-    @Nullable
-    @Override
-    public IBinder onBind(Intent intent) {
-        return null;
-    }
+        ConnectionFactory.blockingWaitForInitialization();
```

Comment:
...probably that one? As it now does not handle async things anymore?

Reply:
My main problem was the approach of ContinuingIntentService, that is, letting the the service run indefinitely due to the async HTTP requests, which made the IntentService useless anyway.
Now I'm needing it for the blocking wait, thus I converted the HTTP to be sync requests and everything is in order :-)

Reply:
Ah, ok, thanks for the explanation! :)